lib/request_overrider.js

```diff
@@ -313,184 +316,188 @@ function RequestOverrider(req, options, interceptors, remove, cb) {
 
       }
     }
+    return continueWithResponseBody(responseBody);
 
-    //  Transform the response body if it exists (it may not exist if we have `responseBuffers` instead)
-    if(responseBody) {
-      debug('transform the response body');
 
-      if (!Buffer.isBuffer(responseBody) && !isStream(responseBody)) {
-        if (typeof responseBody === 'string') {
-          responseBody = new Buffer(responseBody);
-        } else {
-          responseBody = JSON.stringify(responseBody);
+    function continueWithResponseBody(responseBody) {
```

Comment:
After this line I didn't touch the code just indent it because of the new function definition.